app/controllers/api/v1/transports_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Api
+  module V1
+    class TransportsController < Api::V1::ApiController
+
+      skip_authorization_check only: :update_hook
+      load_and_authorize_resource :transport_order, parent: false, except: [:update_hook]
+
+      api :GET, '/v1/transports/providers', "List all GoodCity Tranports Options."
+      def providers
+        render json: TransportProvider.all.cached_json
+      end
+
+      api :POST, '/v1/transports/quote', "Get provider quote"
+      param :provider, String, desc: "Provider selected for transport"
+      param :vehicle_type, String, desc: "Transport vehicle-type"
+      param :offer_id, String, desc: "Id of the offer"
+      param :schedule_at, String, desc: "Scheduled time for delivery"
+      param :district_id, String, desc: "Id of the district"
+      def quote
+        order_price = TransportService.new(transport_params.to_h).quotation
+        render json: order_price
+      end
+
+      api :POST, '/v1/transports/book', "Book transport"
+      param :provider, String, desc: "Provider selected for transport"
+      param :vehicle_type, String, desc: "Transport vehicle-type"
+      param :offer_id, String, desc: "Id of the offer"
+      param :schedule_at, String, desc: "Scheduled time for delivery"
+      param :district_id, String, desc: "Id of the district"
+      param :pickup_contact_name, String, desc: "Contact Person Name"
+      param :pickup_contact_phone, String, desc: "Contact Person Mobile"
+      param :pickup_street_address, String, desc: "Pickup Address"
+      def book
+        order_info = TransportService.new(transport_params.to_h).book
```

Comment:
Can you filter by `offer_id` to ensure user can only book for their offers (or admin can book on their behalf)?